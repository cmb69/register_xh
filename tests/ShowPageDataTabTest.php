<?php

/**
 * Copyright (c) 2012-2021 Christoph M. Becker
 */

namespace Register;

use PHPUnit\Framework\TestCase;
use ApprovalTests\Approvals;

use Register\Infra\Request;
use Register\Infra\Url;
use Register\Infra\View;

class ShowPageDataTabTest extends TestCase
{
    public function testRendersPageDataTab(): void
    {
        $text = XH_includeVar("./languages/en.php", "plugin_tx")["register"];
        $sut = new ShowPageDataTab($text, new View("./", $text));
        $request = $this->createStub(Request::class);
        $request->expects($this->any())->method("url")->willReturn(new Url("/", "SomePage"));
        $request->expects($this->any())->method("coreStyleFolder")->willReturn("../../assets/css/");
        $response = $sut($request, ["register_access" => "cmb"]);
        Approvals::verifyHtml($response->output());
    }
}
