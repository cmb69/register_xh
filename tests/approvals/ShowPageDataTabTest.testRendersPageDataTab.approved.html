
<form action="./?SomePage" method="post" id="register">
  <p>
    <div class="pl_tooltip">
      <img src="../../assets/css/help_icon.png" alt="Question mark">
      <div>A comma separated list of user groups which have access to this page. Leave empty to grant access to everybody, including visitors.</div>
    </div>
    <label>
      Access Groups<br/>
      <input name="register_access" value="cmb">
    </label>
  </p>
  <input name="save_page_data" type="hidden">
  <p>
    <button>Submit</button>
  </p>
</form>
